File: tools/analyze_excel.py

import pandas as pd
import os

file_path = "TOTAL AT 計算表（50lvシンク）.xlsx"

if not os.path.exists(file_path):
    print(f"Error: File not found at {file_path}")
    exit(1)

try:
    # Load the Excel file to get sheet names
    xl = pd.ExcelFile(file_path)
    print(f"Sheet names: {xl.sheet_names}")

    # Read the first sheet (assuming it's the main one)
    # If there are multiple relevant sheets, we might need to iterate
    for sheet_name in xl.sheet_names:
        print(f"\n--- Sheet: {sheet_name} (First 10 rows) ---")
        df = xl.parse(sheet_name)
        # Display all columns
        pd.set_option('display.max_columns', None)
        pd.set_option('display.width', 1000)
        print(df.head(10))
        print("\nColumn Types:")
        print(df.dtypes)

except Exception as e:
    print(f"An error occurred: {e}")

File: tools/analyze_excel.js

const XLSX = require('xlsx');
const fs = require('fs');

const filePath = 'TOTAL AT 計算表（50lvシンク）.xlsx';

if (!fs.existsSync(filePath)) {
    console.error(`File not found: ${filePath}`);
    process.exit(1);
}

try {
    const workbook = XLSX.readFile(filePath);
    console.log('Sheet Names:', workbook.SheetNames);

    workbook.SheetNames.forEach(sheetName => {
        console.log(`\n--- Sheet: ${sheetName} ---`);
        const worksheet = workbook.Sheets[sheetName];
        // Convert to JSON to see the data structure, limit to first 10 rows
        const data = XLSX.utils.sheet_to_json(worksheet, { header: 1 });
        data.slice(0, 10).forEach(row => {
            console.log(JSON.stringify(row));
        });
    });

} catch (e) {
    console.error('Error reading file:', e);
}
